src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAOSpec.scala

```diff
@@ -63,4 +65,32 @@ class HttpGoogleServicesDAOSpec extends FlatSpec with Matchers {
     objectMetadata.name should equal("test-composite-object")
     objectMetadata.md5Hash should equal(None)
   }
+
+  {
+    // https://stackoverflow.com/a/24375762/818054
+    val updatePreservingOrder = PrivateMethod('updatePreservingOrder)
```

Comment:
Does `updatePreservingOrder` have to be private? Or can you extract the thing that you want to test into a separate object that this can logically include in its public API?

Reply:
Good question. The public API is `updateSpreadsheet`, above, which interacts directly with Google - so we can't conveniently test that.

I'd entertain making `updatePreservingOrder` `public` to make tests simpler, but it doesn't naturally belong to the public API of `HttpGoogleServicesDAO`.

This is the first time I've encountered this specific problem in Scala, and decided to explore what's possible in `ScalaTest`.

Reply:
Another thought is to pull the method into a supporting trait that would allow for easier testing.